integration/smartbft/smartbft_test.go

```diff
@@ -1662,6 +1662,159 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
 			By("invoking the chaincode, again")
 			invokeQuery(network, peer, network.Orderers[numberKill], channel, 80)
 		})
+
+		It("smartbft the leader froze, waiting for an answer", func() {
+			networkConfig := nwo.MultiNodeSmartBFT()
+			networkConfig.Channels = nil
+			channel := "testchannel1"
+
+			network = nwo.New(networkConfig, testDir, client, StartPort(), components)
+			network.GenerateConfigTree()
+			network.Bootstrap()
+
+			var ordererRunners []*ginkgomon.Runner
+			for _, orderer := range network.Orderers {
+				runner := network.OrdererRunner(orderer, "FABRIC_LOGGING_SPEC=orderer.consensus.smartbft=debug:grpc=debug")
+				ordererRunners = append(ordererRunners, runner)
+				proc := ifrit.Invoke(runner)
+				ordererProcesses = append(ordererProcesses, proc)
+				Eventually(proc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			}
+
+			peerGroupRunner, _ := peerGroupRunners(network)
+			peerProcesses = ifrit.Invoke(peerGroupRunner)
+			Eventually(peerProcesses.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			peer := network.Peer("Org1", "peer0")
+
+			joinChannel(network, channel)
+
+			By("Waiting for followers to see the leader")
+			Eventually(ordererRunners[1].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[2].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[3].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+
+			// Stopped 2 out of 4 nodes
+			for i := 2; i <= 3; i++ {
+				orderer := network.Orderers[i]
+				By(fmt.Sprintf("Killing %s", orderer.Name))
+				ordererProcesses[i].Signal(syscall.SIGTERM)
+				Eventually(ordererProcesses[i].Wait(), network.EventuallyTimeout).Should(Receive())
+			}
```

Comment:
Why do you need to start four nodes to shut down 2, send a command to the first one, and then start the other two back again? 

Can't you start only two? And could you do the same without restarting? My understanding of the case should be:

1. The network is being initialized, but not the entire cluster has started yet
2. Non-BFT client sends transaction only to the leader; the rest are not aware, and some nodes have not even started yet
3. The leader is waiting to get prepared responses, while the rest are not even aware he is waiting 
4. No one can wake up the leader from waiting; the rest of the cluster is not aware of the transaction being in flight

Therefore, I do not understand why you cannot simulate the same behavior in the test and need this extra step of starting, stoping and restarting again.

Reply:
You're right, I'll redo it.

Reply:
rewritten